src/pages/application_status/index.tsx

```diff
@@ -0,0 +1,373 @@
+import { useState } from 'react'
+import Head from 'next/head'
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react'
+import Image from 'next/image'
+import TopNavbar from '~/components/common/TopNavbar'
+import withAuth, { BaseProps } from '~/utils/withAuth'
+
+enum PageState {
+  LOGIN,
+  FORGET_PASSWORD,
+}
+
+const Application_Status: React.FC<BaseProps> = ({ session }) => {
+  //this part needs backend to update the status
+  const [isOpen, setIsOpen] = useState(false)
+  const [isAcceptOpen, setIsAcceptOpen] = useState(false)
+  const [isRejectOpen, setIsRejectOpen] = useState(false)
+  const onOpenAccept = () => setIsAcceptOpen(true)
+  const onCloseAccept = () => setIsAcceptOpen(false)
+  const onOpenReject = () => setIsRejectOpen(true)
+  const onCloseReject = () => setIsRejectOpen(false)
+  const onOpen = () => setIsOpen(true)
+  const onClose = () => setIsOpen(false)
```

Comment:
Perhaps to make it neater, you can do like

```suggestion
  const toggleOpen = () => setIsOpen(!isOpen)
```

but current implementation is fine